### ISSUE TITLE
[BUG] infinite loop on windows: searching for files

### ISSUE THREAD

**Issue by bkreider** [Anaconda-Platform/anaconda-project#379]
#### ALL software version info

version 0.10.1

#### Description of expected behavior and the observed behavior

On windows there are two functions that recursively look in parent directories.  Because the windows filesystem is case insensitive, it can get trapped at c:\ forever looping.  These two functions suffer from the problem:

  * local_state_file.py:42:def load_for_directory
  * project_lock_file.py:45:def load_for_directory
  
When it gets to C:\ it does a string compare that looks like this:  "c:\ == C:\".  Since this is never true, it gets trapped in  an infinite loop.
```
       current_dir = directory
        while current_dir != os.path.realpath(os.path.dirname(current_dir)):
            for name in possible_project_lock_file_names:
                path = os.path.join(current_dir, name)
                if os.path.isfile(path):
                    return ProjectLockFile(path)

            if scan_parents:
                current_dir = os.path.dirname(os.path.abspath(current_dir))
                continue
            else:
                break
```

**Comment by AlbertDeFusco** [comment]
I believe this PR will fix the issue: https://github.com/Anaconda-Platform/anaconda-project/pull/380

**Comment by mcg1969** [comment]
Thanks for the careful diagnosis — the analysis was spot on. (PR #380, despite the title, didn't actually touch these loops.)

I've fixed this in #433. The new code walks parents using a `dirname` fixed-point check on an `abspath`-normalized input, which terminates regardless of case-insensitivity or symlinks. The same buggy idiom existed in three places (`local_state_file.py`, `project_lock_file.py`, and `project_file.py`); all three are fixed. A regression test that hangs under the old code and passes under the new is included.

I'll close this once #433 lands. For broader context on the project's direction:

* Anaconda Project has largely been superseded by [pixi](https://pixi.sh/), [uv](https://docs.astral.sh/uv/), and [conda-workspaces](https://github.com/conda-incubator/conda-workspaces).
* See the [Pixi support](https://anaconda-project.readthedocs.io/en/latest/pixi-support.html) page for the `anaconda-project export-pixi` migration tool.
* The `anaconda-project` package itself remains available; existing projects on its stable feature set continue to be supported, but new enhancement work is generally out of scope.
